electrum_nmc/electrum/names.py

```diff
@@ -132,6 +132,317 @@ def validate_value_length(value: bytes):
+                # First 61 characters of the gTLD
+                + r"+[A-Za-z0-9][A-Za-z0-9-_]{0,61}"
+                # Last character of the gTLD
+                + rf"[A-Za-z]{r'.$' if rfc_1034 else r'$'}",
+                value.encode("idna").decode("utf-8"),
+                re.IGNORECASE,
+            )
+        except UnicodeError:
+            return False
+
+def validate_domains(domain: str) -> None:
+    if domain.endswith('.'):
+        relative_domain = domain[:-1]
+        if not validate_relative_domain(relative_domain):
+            raise ValueError(f'Invalid domain: {domain}')
+
+    else:
+        if not validate_relative_domain(domain):
+            raise ValueError(f'Invalid domain: {domain}')
+
+        try:
+            # Read the TLD set from the JSON file
+            current_directory = os.path.dirname(os.path.abspath(__file__))
+            json_file_path = os.path.join(current_directory, 'tld_list.json')
+
+            with open(json_file_path, 'r') as f:
+                tld_set = set(json.load(f))
+        except Exception as e:
+            raise ValueError(f"Unable to read the TLD list: {e}")
+
+        TLD = domain.split('.')[-1].lower()
+        if tld_set and TLD.upper() in tld_set:
+            raise ValueError(f'Recognized TLD which might indicate an invalid domain: {domain}')
+
+def validate_nameservers(domain: str) -> None:
+    if domain.endswith('.'):
+        relative_domain = domain[:-1]
+        if not validate_relative_domain(relative_domain):
+            raise ValueError(f'Invalid domain: {domain}')
+
+    else:
+        if not validate_relative_domain(domain):
+            raise ValueError(f'Invalid domain: {domain}')
+
+        try:
+            # Read the TLD set from the JSON file
+            current_directory = os.path.dirname(os.path.abspath(__file__))
+            json_file_path = os.path.join(current_directory, 'tld_list.json')
+
+            with open(json_file_path, 'r') as f:
+                tld_set = set(json.load(f))
+        except Exception:
+            # If the JSON file is not found, we will skip the TLD validation
+            return
+
+        TLD = domain.split('.')[-1].lower()
+        if tld_set and TLD.upper() not in tld_set:
+            raise ValueError(f'Unrecognized TLD which might indicate an invalid domain: {domain}')
+
+def validate_DNSSEC(Hash: str, HashType: int, Algorithm: int) -> None:
+    # DNSSEC Algorithm Numbers: https://www.iana.org/assignments/dns-sec-alg-numbers/dns-sec-alg-numbers.xhtml
+    # DS RR Types (Hash Types): https://www.iana.org/assignments/ds-rr-types/ds-rr-types.xhtml#ds-rr-types-1
+    try:
+        # Decode the base64 hash to ensure it's valid base64
+        if Hash:
+            a2b_base64(Hash)
+
+        # Validate HashType
+        if HashType:
+            match HashType:
+                case 1:
+                    raise Exception("Warning: SHA-1 (HashType 1) is considered weak due to vulnerabilities to collision attacks.")
```

Comment:
This is more info than the user needs; maybe reword to `Insecure hash algorithm.`?